File: api/app/models/ingredientes.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class IngredienteCreate(BaseModel):
    nombre: str = Field(min_length=2, max_length=150)
    unidad: str = Field(min_length=1, max_length=20)
    stock_actual: Decimal = Field(default=Decimal("0"), ge=0)
    stock_minimo: Decimal = Field(ge=0)
    costo_unitario: Decimal = Field(gt=0)
    activo: bool = True


class IngredienteOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    nombre: str
    unidad: str
    stock_actual: Decimal
    stock_minimo: Decimal
    costo_unitario: Decimal
    activo: bool


class ActualizarStock(BaseModel):
    cantidad: Decimal = Field(description="Delta a aplicar al stock actual; puede ser negativo")

File: mobile/api/ingredientes.js
import { request } from './client';

export function getIngredientes() {
  return request('/ingredientes');
}

export function createIngrediente({ nombre, unidad, stockMinimo, costoUnitario, stockInicial }) {
  return request('/ingredientes', {
    method: 'POST',
    body: {
      nombre,
      unidad,
      stock_actual: stockInicial || 0,
      stock_minimo: stockMinimo,
      costo_unitario: costoUnitario,
      activo: true,
    },
  });
}

export function ajustarStock(id, cantidad) {
  return request(`/ingredientes/${id}/stock`, {
    method: 'PUT',
    body: { cantidad },
  });
}

export function deleteIngrediente(id) {
  return request(`/ingredientes/${id}`, { method: 'DELETE' });
}
